### ISSUE TITLE
Only the first two Resource Duplicate are found - E0000

### ISSUE THREAD

**Issue by asantos-fuze** [aws-cloudformation/cfn-lint#1906]
*cfn-lint version: (`cfn-lint 0.45.0`) Build from source: Commit: 002b3d1*

*Description of issue.*

This issue was 1st described on #1868 and fixed on #1900 

I was validating the fix and it works if there are two duplicated resources, but not if you have 3 or more duplicated resources.

I get the error `E0000 Duplicate resource found "DevRouteTableRoute2" (line 18)`  and `E0000 Duplicate resource found "DevRouteTableRoute2" (line 13)` when I try to validate a CFn template with duplicated resources. I **should get has many errors has the number of duplicated resources on the template**. The template had more than one duplication (in this case 5) , but the linter only showed the 1st and 2nd.
Please provide as much information as possible:
* Test template:
```yaml
---
AWSTemplateFormatVersion: "2010-09-09"

Resources:

  DevRouteTableRoute1:
    Type: AWS::EC2::TransitGatewayRoute
    Properties:
      Blackhole: true
      DestinationCidrBlock: 10.1.1.0/24
      TransitGatewayRouteTableId: tgw-rtb-d65d85280f2c9e51c
  DevRouteTableRoute2:
    Type: AWS::EC2::TransitGatewayRoute
    Properties:
      DestinationCidrBlock: 10.1.2.0/24
      TransitGatewayAttachmentId: tgw-attach-9bdf1fdfb44d9c7d7
      TransitGatewayRouteTableId: tgw-rtb-d65d85280f2c9e51c
  DevRouteTableRoute2:
    Type: AWS::EC2::TransitGatewayRoute
    Properties:
      DestinationCidrBlock: 10.1.3.0/24
      TransitGatewayAttachmentId: tgw-attach-b16ea962254d4373e
      TransitGatewayRouteTableId: tgw-rtb-d65d85280f2c9e51c
  DevRouteTableRoute2:
    Type: AWS::EC2::TransitGatewayRoute
    Properties:
      DestinationCidrBlock: 10.1.4.0/24
      TransitGatewayAttachmentId: tgw-attach-b16ea962254d4373e
      TransitGatewayRouteTableId: tgw-rtb-d65d85280f2c9e51c

  ProdRouteTableRoute1:
    Type: AWS::EC2::TransitGatewayRoute
    Properties:
      Blackhole: true
      DestinationCidrBlock: 10.1.1.0/24
      TransitGatewayRouteTableId: tgw-rtb-5e9891f700a3af932
  ProdRouteTableRoute2:
    Type: AWS::EC2::TransitGatewayRoute
    Properties:
      DestinationCidrBlock: 10.1.2.0/24
      TransitGatewayAttachmentId: tgw-attach-9bdf1fdfb44d9c7d7
      TransitGatewayRouteTableId: tgw-rtb-5e9891f700a3af932
  ProdRouteTableRoute2:
    Type: AWS::EC2::TransitGatewayRoute
    Properties:
      DestinationCidrBlock: 10.1.3.0/24
      TransitGatewayAttachmentId: tgw-attach-b16ea962254d4373e
      TransitGatewayRouteTableId: tgw-rtb-5e9891f700a3af932
  ProdRouteTableRoute4:
    Type: AWS::EC2::TransitGatewayRoute
    Properties:
      DestinationCidrBlock: 10.1.4.0/24
      TransitGatewayAttachmentId: tgw-attach-b16ea962254d4373e
      TransitGatewayRouteTableId: tgw-rtb-5e9891f700a3af932
```
  * Validation Output
 ```
E0000 Duplicate resource found "DevRouteTableRoute2" (line 18)
template.yml:18:3

E0000 Duplicate resource found "DevRouteTableRoute2" (line 13)
template.yml:13:5
```
 * Expected Output
```
E0000 Duplicate resource found "ProdRouteTableRoute2" (line 43)
template.yml:43:3

E0000 Duplicate resource found "ProdRouteTableRoute2" (line 37)
template.yml:37:3

E0000 Duplicate resource found "DevRouteTableRoute2" (line 24)
template.yml:24:3

E0000 Duplicate resource found "DevRouteTableRoute2" (line 18)
template.yml:18:3

E0000 Duplicate resource found "DevRouteTableRoute2" (line 13)
template.yml:13:5
```

We should get has many errors has the number of duplicated resources on the template. This is how the linter is working for the other errors

cc/ @kddejong

**Comment by kddejong** [comment]
Yea agreed.  I'm going to leave this open for tracking. Since we handle these errors in the parser we are somewhat subject to the parser can do for passing and handling errors.  I think I can do this but it may take a little work to get it working the right way.
